apps/api/src/app/workflows-v2/shared/schemas/digest-control.schema.ts

```diff
@@ -67,7 +67,7 @@ export const digestUiSchema: UiSchema = {
     },
     digestKey: {
       component: UiComponentEnum.DIGEST_KEY,
-      placeholder: null,
+      placeholder: '',
```

Comment:
the UI schema placeholder for the digest key can't be null, otherwise react warning will be raised

Reply:
If I do this:

1. Create new workflow
2. Add digest step - it will show default 30 seconds in the drawer input field, I leave it as is

Where do I see the default 30 seconds being saved as control value?

Its not returned from `GET step` endpoints in the `data.controls.values`
Its not present in database `controls` collection

Unless I change the value in to force auto-save.

I also opened this topic a while ago [here], (https://novu.slack.com/archives/C07HVLG7G2U/p1732895403877039?thread_ts=1732887674.583079&cid=C07HVLG7G2U), might be possible that I'm missing something important here.

Reply:
If I do this:

1. Create new workflow
2. Add digest step - it will show default 30 seconds in the drawer input field (note: it doesn't show as a placeholder, but a value), I leave it as is

![image](https://github.com/user-attachments/assets/986bb792-f789-4f71-bd1d-f390ebdc66e2)


Where do I see the default 30 seconds being saved as control value?

Its not returned from `GET step` endpoints in the `data.controls.values`
Its not present in database `controls` collection

Unless I change the value in the input field to force auto-save.

I also opened this topic a while ago [here](https://novu.slack.com/archives/C07HVLG7G2U/p1732895403877039?thread_ts=1732887674.583079&cid=C07HVLG7G2U), might be possible that I'm missing something important here.

Reply:
@ChmaraX the `30 seconds` are the "default values" you get from the `uiSchema`. It's not the same as a `placeholder` in the UI inputs, the name is "unfortunate".